src/server/middleware/auth.ts

```diff
@@ -278,6 +282,11 @@ const handleKeypairAuth = async (
       throw error;
     }
 
+    if (data && req.method === "POST" && data !== hashRequestBody(req)) {
+      error = "The request body has been tampered with.";
```

Comment:
This error sounds scary when the dev may have just made a mistake. Let's make sure errors directly state the issue and optionally provide a way to fix it:

> The request body does not match the hash in the access token. See: https://portal.thirdweb.com/engine/features/keypair-authentication

Reply:
Makes sense.

can we tinyurl this?